pyked/chemked.py

```diff
@@ -233,6 +241,238 @@ def get_dataframe(self, output_columns=None):
+        if exists(filename) and not overwrite:
+            raise NameError(filename + ' already present. Specify "overwrite=True" '
+                            'to overwrite, or rename.'
+                            )
+
+        with open(filename, 'w') as yaml_file:
+            yaml.dump(self._properties, yaml_file)
+
+    def convert_to_ReSpecTh(self, filename):
+        """Convert ChemKED record to ReSpecTh XML file.
+
+        This converter uses common information in a ChemKED file to generate a
+        ReSpecTh XML file. Note that some information may be lost, as ChemKED stores
+        some additional attributes.
+
+        Arguments:
+            filename (`str`): Filename for output ReSpecTh XML file.
+
+        Example:
+            >>> dataset = ChemKED(yaml_file)
+            >>> dataset.convert_to_ReSpecTh(xml_file)
+        """
+        root = etree.Element('experiment')
+
+        file_author = etree.SubElement(root, 'fileAuthor')
+        file_author.text = self.file_author['name']
+
+        # right now ChemKED just uses an integer file version
+        file_version = etree.SubElement(root, 'fileVersion')
+        major_version = etree.SubElement(file_version, 'major')
+        major_version.text = str(self.file_version)
+        minor_version = etree.SubElement(file_version, 'minor')
+        minor_version.text = '0'
+
+        respecth_version = etree.SubElement(root, 'ReSpecThVersion')
+        major_version = etree.SubElement(respecth_version, 'major')
+        major_version.text = '1'
+        minor_version = etree.SubElement(respecth_version, 'minor')
+        minor_version.text = '0'
+
+        # Only ignition delay currently supported
+        exp = etree.SubElement(root, 'experimentType')
+        if self.experiment_type == 'ignition delay':
+            exp.text = 'Ignition delay measurement'
+        else:
+            raise NotImplementedError('Only ignition delay type supported for conversion.')
+
+        reference = etree.SubElement(root, 'bibliographyLink')
+        citation = ''
+        for author in self.reference.authors:
+            citation += author['name'] + ', '
+        citation += (self.reference.journal + ' (' + str(self.reference.year) + ') ' +
+                     str(self.reference.volume) + ':' + self.reference.pages + '. ' +
+                     self.reference.detail
+                     )
+        reference.set('preferredKey', citation)
+        reference.set('doi', self.reference.doi)
+
+        apparatus = etree.SubElement(root, 'apparatus')
+        kind = etree.SubElement(apparatus, 'kind')
+        kind.text = self.apparatus.kind
+
+        common_properties = etree.SubElement(root, 'commonProperties')
+        # ChemKED objects have no common properties once loaded. Check for properties
+        # among datapoints that tend to be common
+        common = []
+        composition = self.datapoints[0].composition
+        # Composition type *has* to be the same
```

Comment:
> This particular location is testing whether all of the datapoints have the same composition, so I think it's appropriate here (if the others don't match datapoint 0, then it isn't a common property- good!)

Is this checking that the `composition_type`s are all the same? If not, I think it should because, hypothetically anyways, the species and numbers could all be equal (i.e., the `composition`), but some could be mass fraction and some could be mole fraction (i.e., the `composition_type`s could be different). This isn't likely to happen, but I think we should code more defensively because you never know what those crazy experimentalists are going to try to do next...

In addition, the comment isn't correct for a CK instance... not sure if you meant it to refer to the CK class or the ReSpecTh file. Maybe that has to be one of the cases we can't convert to ReSpecTh, if there are multiple `composition_type`s in a set of datapoints.